core/tauri/src/api/process/command.rs

```diff
@@ -406,9 +421,14 @@ fn spawn_pipe_reader<F: Fn(String) -> CommandEvent + Send + Copy + 'static>(
             break;
           }
           let tx_ = tx.clone();
-          let line = match character_encoding {
-            Some(encoding) => Ok(encoding.decode_with_bom_removal(&buf).0.into()),
-            None => String::from_utf8(buf.clone()),
+          let line = match encoding {
+            EncodingWrapper::Text(character_encoding) => match character_encoding {
+              Some(encoding) => Ok(Buffer::Text(
+                encoding.decode_with_bom_removal(&buf).0.into(),
+              )),
+              None => String::from_utf8(buf.clone()).map(|text| Buffer::Text(text)),
+            },
+            EncodingWrapper::Raw => Ok(Buffer::Raw(buf.clone())),
```

Comment:
maybe try always sending raw data here and move the decoding logic to `endpoints/shell.rs` right before passing it to Javascript